function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -61,7 +61,7 @@ func (this *grpcDispatcher) handleIncoming() {
 				err := this.stream.Send(grpcMessage)
 				if err != nil {
 					if streamClosureDiagnosed(err) {
-						this.closed <- true
+						this.closed <- *new(struct{})
```

Comment:
Rather than sending to the channel, the channel should be closed. That's the normal way to send a signal using a channel.